src/StreamrClient.js

```diff
@@ -476,9 +478,28 @@ export default class StreamrClient extends EventEmitter {
             this._requestSubscribe(sub)
 
             // Once subscribed, ask for a resend
-            sub.once('subscribed', () => {
+            sub.once('subscribed', async () => {
                 if (sub.hasResendOptions()) {
-                    this._requestResend(sub)
+                    await this._requestResend(sub)
+                    // the requested messages might not have been stored yet, so retry if no answer after 'storageDelay' seconds
+                    const secondResend = setTimeout(() => this._requestResend(sub), this.options.storageDelay)
+                    // once a message is received, gap filling in Subscription.js will check if this satisfies the resend and request
+                    // another resend if it doesn't. So we can anyway clear this resend request.
+                    const callback = () => clearTimeout(secondResend)
+                    const handler = () => {
+                        callback()
+                        const removeCallbacks = () => {
+                            sub.removeListener('message received', handler)
+                            sub.removeListener('no_resend', handler)
+                            sub.removeListener('unsubscribed', handler)
+                            sub.removeListener('error', handler)
+                        }
+                        removeCallbacks()
+                    }
+                    sub.once('message received', handler)
+                    sub.once('no_resend', handler)
```

Comment:
`no_resend` shouldn't actually cancel the retry! For example, let's assume there's 1 message in the stream and it's still in "storage delay":

- resend last 1
- `no_resend` (message not written to storage yet)
- cancel retry

The client wouldn't ever get the message. Instead if it retries:

- resend last 1
- `no_resend` (message not written to storage yet)
- wait for `storageDelay`
- resend last 1
- now the message is found and delivered